tests/ops/op_math/test_parametrized_hamiltonian.py

```diff
@@ -0,0 +1,370 @@
+
+    def test_H_parametrized(self):
+        """Test H_parametrized is a function that, when passed parameters,
+        returns an Operator of the expected format."""
+        H_param = test_example.H_parametrized
+
+        H = H_param([1, 2], 0.5)
+        assert len(H) == 2
+        assert isinstance(H, qml.ops.Sum)
+        assert isinstance(H[0], qml.ops.SProd)
+        assert isinstance(H[1], qml.ops.SProd)
+
+    def test__repr__(self):
+        """Test repr method returns expected string"""
+        str = repr(test_example)
+        assert str == "ParametrizedHamiltonian: terms=4"
+
+    def test_wire_attribute(self):
+        """Tests that the wires attribute contains the expected wires, in the expected order"""
+        coeffs = [1, f1, 2, f2]
+        ops = [qml.PauliX(3), qml.PauliX(2), qml.PauliX(0), qml.PauliX(1)]
+
+        H = ParametrizedHamiltonian(coeffs, ops)
+        assert H.wires == H(param, 2).wires
+        assert H.wires == Wires([3, 0, 2, 1])
+
+
+class TestCall:
+
+    coeffs_and_ops_and_params = (
+        (
+            [f1, f2],
+            [qml.PauliX(0), qml.PauliY(1)],
+            [1, 2],
+        ),  # no H_fixed term, multiple H_parametrized terms
+        ([f1], [qml.PauliX(0)], [1]),  # no H_fixed term, one H_parametrized term
+        (
+            [1.2, 2.3],
+            [qml.PauliX(0), qml.PauliY(1)],
+            [],
+        ),  # no H_parametrized term, multiple H_fixed terms
+        ([1.2], [qml.PauliX(0)], []),  # no H_parametrized term, one H_fixed term
+        ([1, f1], [qml.PauliX(3), qml.PauliY(2)], [1]),  # one of each
+        ([1.2, f1, 2.3, f2], [qml.PauliX(0) for i in range(4)], [1, 2]),  # multiples of each
+    )
+
+    @pytest.mark.parametrize("coeffs, ops, params", coeffs_and_ops_and_params)
+    def test_call_succeeds_for_different_shapes(self, coeffs, ops, params):
+        """Test that calling the ParametrizedHamiltonian succeeds for
+        different numbers of H_fixed and H_parametrized terms"""
+        pH = ParametrizedHamiltonian(coeffs, ops)
+        pH(params, 0.5)
+
+    def test_call_returns_expected_results(self):
+        """Test result of calling the ParametrizedHamiltonian makes sense"""
+        pH = ParametrizedHamiltonian([1.2, f1, 2.3, f2], [qml.PauliX(i) for i in range(4)])
+        params = [4.5, 6.7]
+        t = 2
+        op = pH(params, t)
+
+        assert isinstance(op, qml.ops.Sum)
+        assert len(op) == 2
+
+        H_fixed = op[0]
+        H_parametrized = op[1]
+        expected_H_fixed = qml.op_sum(
+            qml.s_prod(1.2, qml.PauliX(0)), qml.s_prod(2.3, qml.PauliX(2))
+        )
+        expected_H_parametrized = qml.op_sum(
+            qml.s_prod(f1(params[0], t), qml.PauliX(1)), qml.s_prod(f2(params[1], t), qml.PauliX(3))
+        )
+
+        assert qml.equal(H_fixed, expected_H_fixed)
+        assert qml.equal(H_parametrized, expected_H_parametrized)
+
+    def test_call_with_qutrit_operators(self):
```

Comment:
Cool that this works already! Missing doc string btw (though it is not required according to our pylintrc for tests)

Reply:
added :)